### PR TITLE
fix: remove stripe metadata and more logos

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Team metadata was exposing stripe ids, and also in org context, subteams was relying on logo instead of avatar.png URL, which increased the page size by a lot. And teams were not updating logo when saving.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Visiting org profile should load all subteams, with their avatar and a considerable reduced `NEXT_DATA` object when running `yarn build` and `yarn start`

### CODE REVIEW

File: apps/web/components/team/screens/Team.tsx [hunk 1]
```diff
@@ -58,10 +57,10 @@ const Members = ({ members, teamName }: { members: MemberType[]; teamName: strin
...
 
-const Team = ({ team }: { team: TeamTypeWithSafeHtml }) => {
+const Team = ({ members, teamName }: { members: MemberType[]; teamName: string | null }) => {
   return (
     <div>
-      <Members members={team.members} teamName={team.name} />
+      <Members members={members} teamName={teamName} />
     </div>
   );
 };
```
Comment (leog) [inline apps/web/components/team/screens/Team.tsx:66]: Fixing the type for expected team as we now removed properties from team metadata (stripe ids)

File: apps/web/pages/team/[slug].tsx [hunk 2]
```diff
@@ -161,14 +159,8 @@ function TeamPage({ team, isUnpublished, markdownStrippedBio, isValidOrgDomain }
...
-            <h2 className="font-cal text-emphasis mb-2 text-3xl">
-              {" " + t("org_no_teams_yet", { defaultValue: "This organization has no teams yet" })}
-            </h2>
-            <p className="text-emphasis mx-auto max-w-md">
-              {t("org_no_teams_yet_description", {
-                defaultValue: "If you are an administrator, be sure to create teams to be shown here.",
-              })}
-            </p>
+            <h2 className="font-cal text-emphasis mb-2 text-3xl">{" " + t("org_no_teams_yet")}</h2>
+            <p className="text-emphasis mx-auto max-w-md">{t("org_no_teams_yet_description")}</p>
```
Comment (leog) [inline apps/web/pages/team/[slug].tsx:163]: Removed default values introduced recently as 99% of the extra page size came from logo not i18n

File: apps/web/pages/team/[slug].tsx [hunk 3]
```diff
@@ -281,9 +273,7 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
     isTeamView: true,
     isOrgView: isValidOrgDomain && context.resolvedUrl === "/",
   });
-  const ssr = await ssrInit(context, {
-    noI18nPreload: isValidOrgDomain && !team?.parent,
-  });
+  const ssr = await ssrInit(context);
```
Comment (leog) [inline apps/web/pages/team/[slug].tsx:276]: Removed option to not preload i18n introduced recently as 99% of the extra page size came from logo not i18n

File: packages/features/ee/teams/pages/team-profile-view.tsx [hunk 4]
```diff
@@ -167,6 +167,7 @@ const ProfileView = () => {
                     name: values.name,
                     slug: values.slug,
                     bio: values.bio,
+                    logo: values.logo,
```
Comment (leog) [inline packages/features/ee/teams/pages/team-profile-view.tsx:170]: Teams' logo were not being updated as it wasn't counted as a change in the form before submitting

File: packages/lib/server/queries/teams/index.ts [hunk 5]
```diff
@@ -64,13 +64,11 @@ export async function getTeamWithMembers(args: {
         id: true,
         slug: true,
         name: true,
-        logo: true,
       },
     },
     children: {
       select: {
         name: true,
-        logo: true,
```
Comment (leog) [inline packages/lib/server/queries/teams/index.ts:73]: Bye logo

File: packages/lib/server/queries/teams/index.ts [hunk 6]
```diff
@@ -168,9 +166,18 @@ export async function getTeamWithMembers(args: {
...
   const { inviteTokens, ...teamWithoutInviteTokens } = team;
 
+  // Don't leak stripe payment ids
+  const teamMetadata = teamMetadataSchema.parse(team.metadata);
+  const {
+    paymentId: _,
+    subscriptionId: __,
+    subscriptionItemId: ___,
+    ...restTeamMetadata
+  } = teamMetadata || {};
```
Comment (leog) [inline packages/lib/server/queries/teams/index.ts:179]: Removing stripe-related metadata from team metadata exposed in `NEXT_DATA`

File: packages/ui/components/unpublished-entity/UnpublishedEntity.tsx [hunk 7]
```diff
@@ -22,13 +22,8 @@ export function UnpublishedEntity(props: UnpublishedEntityProps) {
...
         headline={t("team_is_unpublished", {
           team: props.name,
-          defaultValue: `${props.name} is unpublished`,
-        })}
-        description={t(`${props.orgSlug ? "org" : "team"}_is_unpublished_description`, {
-          defaultValue: `This ${
-            props.orgSlug ? "organization" : "team"
-          } link is currently not available. Please contact the organization owner or ask them to publish it.`,
         })}
+        description={t(`${props.orgSlug ? "org" : "team"}_is_unpublished_description`)}
```
Comment (leog) [inline packages/ui/components/unpublished-entity/UnpublishedEntity.tsx:26]: Removed default values introduced recently as 99% of the extra page size came from logo not i18n

**Review by keithwillcode — DISMISSED** [review]
Nice work.

File: packages/features/ee/teams/pages/team-profile-view.tsx [hunk 8]
```diff
@@ -84,7 +84,7 @@ const ProfileView = () => {
   });
 
   const { data: team, isLoading } = trpc.viewer.teams.get.useQuery(
-    { teamId },
+    { teamId, includeTeamLogo: true },
```
Comment (leog) [inline packages/features/ee/teams/pages/team-profile-view.tsx:87]: We need the logo for team profile, so making it possible to retrieve it under a flag to have better control

File: packages/lib/server/queries/teams/index.ts [hunk 9]
```diff
@@ -54,6 +55,7 @@ export async function getTeamWithMembers(args: {
     id: true,
     name: true,
     slug: true,
+    ...(!!includeTeamLogo ? { logo: true } : {}),
```
Comment (leog) [inline packages/lib/server/queries/teams/index.ts:58]: We need the logo for team profile, so making it possible to retrieve it under a flag to have better control

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**